src/override.test.ts

```diff
@@ -0,0 +1,221 @@
+    const sha = await getRootSHA(mockRunCmd);
+    expect(sha).toEqual(rootSHA);
+  });
+
+  it('createOverride returns signable object', async () => {
+    const justification = 'test override';
+    const expiry = Date.now();
+    const mockRunCmd = jest.fn()
+      .mockResolvedValueOnce({ stdout: gpgIdRaw })
+      .mockResolvedValueOnce({ stdout: rootSHA });
+    const override = await createOverride(-10, expiry, justification, mockRunCmd);
+    expect(override.credit).toBe(-10);
+    expect(override.exp).toBe(expiry);
+    expect(override.justification).toBe(justification);
+    expect(override.rootSHA).toBe(rootSHA);
+    expect(override.signedBy).toBe(gpgId);
+  });
+
+  it('importPublicKeys imports all given keys', async () => {
+    const mockRunCmd = jest.fn()
+      .mockResolvedValue({stdout: `
+gpg: keybox './somekeyring.gpg' created
+gpg: key C348A9E00AE60B1C: public key "Trusted User <trusted.user@corp.com>" imported
+gpg: Total number processed: 1
+gpg:               imported: 1
+`});
+
+    const keys = [
+      '/some/path/to/key1.gpg',
+      '/some/path/to/key2.gpg',
+    ];
+
+    await importPublicKeys(keys, mockRunCmd);
+    expect(mockRunCmd).toHaveBeenCalledTimes(2);
+  });
+
+  it('gatherFacts gathers basic Risk Override facts', async () => {
+    const konfig = cloneDeep(config);
+    konfig.flags.pubkeyDir = path.join(__dirname, '../test/fixtures/gpgKeys');
+    konfig.flags.dir = path.join(__dirname, '../test/fixtures');
+
+    const invalidKey = path.join(__dirname, '../test/fixtures/gpgKeys/invalid.gpg');
+    await fs.chmod(invalidKey, '777'); // world writable, hence invalid
+
+    const facts = await gatherFacts(konfig);
+    expect(facts.override.trustedPubKeysDir).toEqual(konfig.flags.pubkeyDir);
+    expect(facts.override.trustedPubKeys.length).toEqual(2);
+    expect(facts.override.repoOverrides.length).toEqual(1);
+  });
+
+  it('validateOverride returns false if Override is expired', async () => {
+    const mockRunCmd = jest.fn();
+    const twoDaysAgo = Date.now() -  (2 * 24 * 60 * 60 * 1000); // in millis
+    const o: Override = {
+      signedBy: 'some user',
+      exp: twoDaysAgo,
+      expires: new Date(twoDaysAgo).toString(),
+      rootSHA: 'abcd',
+      justification: 'test',
+      credit: -1
+    };
+    const isValid = await validateOverride(o, new Date(), mockRunCmd);
+    expect(isValid).toBe(false);
+  });
+
+  it('validateOverride returns false if rootSHA does not match', async () => {
+    const mockRunCmd = jest.fn().mockResolvedValueOnce({ stdout: 'fedcba98766543210' });
+    const twoDaysFromNow = Date.now() +  (2 * 24 * 60 * 60 * 1000); // in millis
+    const o: Override = {
+      signedBy: 'some user',
+      exp: twoDaysFromNow,
+      expires: new Date(twoDaysFromNow).toString(),
+      rootSHA: '0123456789abcdef',
+      justification: 'test',
+      credit: -1
+    };
+    const isValid = await validateOverride(o, new Date(), mockRunCmd);
+    expect(isValid).toBe(false);
```

Comment:
FWIW, this is another way to express these kinds of expectations on Promises.
```suggestion
    await expect(validateOverride(o, new Date(), mockRunCmd)).resolves.toBe(false);
```